aptos-move/block-executor/src/executor.rs

```diff
@@ -179,7 +211,13 @@ where
                     Some(KeyKind::Group(tags)) => tags,
                     Some(KeyKind::Resource) => {
                         return Err(code_invariant_error(format!(
-                            "Group key {:?} recorded as resource KeyKind",
+                            "Group key {:?} recorded as a Resource KeyKind",
```

Comment:
it's unrelated to the pr, but I don't like the big closure, it should be proper function by itself. we may need to pass a few more params, but it'll look cleaner imo